src/crimson/osd/recovery_backend.cc

```diff
@@ -219,60 +219,59 @@ RecoveryBackend::handle_backfill_remove(
 
 RecoveryBackend::interruptible_future<BackfillInterval>
 RecoveryBackend::scan_for_backfill(
-  const hobject_t& start,
+  hobject_t start,
   [[maybe_unused]] const std::int64_t min,
   const std::int64_t max)
 {
   logger().debug("{} starting from {}", __func__, start);
-  auto version_map = seastar::make_lw_shared<std::map<hobject_t, eversion_t>>();
-  return backend->list_objects(start, max).then_interruptible(
-    [this, start, version_map] (auto&& ret) {
-    auto&& [objects, next] = std::move(ret);
-    return seastar::do_with(
-      std::move(objects),
-      [this, version_map](auto &objects) {
-      return interruptor::parallel_for_each(objects,
-	[this, version_map] (const hobject_t& object)
-	-> interruptible_future<> {
-	crimson::osd::ObjectContextRef obc;
-	if (pg.is_primary()) {
-	  obc = pg.obc_registry.maybe_get_cached_obc(object);
+  try {
```

Comment:
As discussed offline,  we can avoid try/catch usage here

Reply:
Done, please take a look again, thanks:-)